### PR TITLE
Feature/infinite scroll

### PR DESCRIPTION
**Pull request by martynaspuskunigis**

### CODE REVIEW

File: src/app/containers/recipes-containers/recipes-container.tsx [hunk 1]
```diff
@@ -21,14 +24,46 @@ class RecipesContainerClass extends React.Component<{}, State> {
...
+            searchKeyword: searchKeyword,
+            currentPage: currentPage,
+            moreRecipes: moreRecipes
         };
     }
 
+    public componentDidMount(): void {
+        window.addEventListener(
+            "scroll",
+            () => {
```
Comment (DeividasBakanas) [inline src/app/containers/recipes-containers/recipes-container.tsx]: Move to separate function.

File: src/app/containers/recipes-containers/recipes-container.tsx [hunk 2]
```diff
@@ -21,14 +24,46 @@ class RecipesContainerClass extends React.Component<{}, State> {
...
+                this.scrollHandler(this.state.searchKeyword, this.state.currentPage);
+            },
+            false
+        );
+    }
+
+    public scrollHandler(searchKeyword: string, currentPage: number): void {
+        if (
+            window.pageYOffset - document.body.scrollHeight === -635 &&
```
Comment (DeividasBakanas) [inline src/app/containers/recipes-containers/recipes-container.tsx]: Move `635` to variable. Don't use `magic` values.

File: src/app/containers/recipes-containers/recipes-container.tsx [hunk 3]
```diff
@@ -40,7 +75,14 @@ class RecipesContainerClass extends React.Component<{}, State> {
                 return <div>Search not initialized...</div>;
             }
             case Abstractions.ItemStatus.Pending: {
-                return <Spinner />;
+                return (
+                    <div className="recipe-list">
+                        {this.state.recipesIds.map((recipeId, index) => (
+                            <RecipesItemContainer key={`recipe-item-${recipeId}-${index}`} recipeId={recipeId} />
```
Comment (DeividasBakanas) [inline src/app/containers/recipes-containers/recipes-container.tsx]: Do you need `index` if `recipeId` provided in `key`?

File: src/app/stores/recipes-store.ts [hunk 4]
```diff
@@ -58,7 +84,10 @@ class RecipesReduceStoreClass extends ReduceStore<StoreState> {
             recipes: [],
             status: Abstractions.ItemStatus.Init,
             activeRecipe: "",
-            favoriteRecipes: []
+            favoriteRecipes: [],
+            currentPage: 1,
+            searchKeyword: "",
+            moreRecipes: true
```
Comment (DeividasBakanas) [inline src/app/stores/recipes-store.ts]: Does this mean `has more recipes`? Maybe should improve naming?

File: src/app/containers/recipes-containers/recipes-container.tsx (hunk 3 above)
Comment (martynaspuskunigis) [reply src/app/containers/recipes-containers/recipes-container.tsx]: Fixed in another pull request

File: src/app/actions/recipes-actions-creators.ts [hunk 5]
```diff
@@ -17,13 +19,28 @@ export namespace RecipesActionsCreators {
...
+            Dispatcher.dispatch(new RecipeIdsLoadFailedAction());
+        }
+    }
+
+    export async function loadMoreRecipes(keyword: string, pageToLoad: number): Promise<void> {
+        Dispatcher.dispatch(new RecipesIdsLoadStartedAction());
+        try {
+            const recipeName = keyword;
+            const apiCall = await fetch(
+                `https://cors-anywhere.herokuapp.com/food2fork.com/api/search?key=${API_KEY}&q=${recipeName}&page=${pageToLoad}`
```
Comment (DeividasBakanas) [inline src/app/actions/recipes-actions-creators.ts]: Make a function to generate API path.

**Pull request merged**